### PR TITLE
Profiles: Add y-axis recalculation based on rank type selection

### PR DESCRIPTION
**Pull request by ghost**
This is by no means perfect, but here are some screenshots:

image

image

image

image

### CODE REVIEW

**Comment by Lekuruu** [comment]
Oh, wow this actually works really well!

File: app/static/js/user.js [hunk 1]
```diff
@@ -704,6 +704,73 @@ function resetUserPerformanceGraph() {
     $rankGraph[0].innerHTML = '';
 }
 
+function getUserPerformanceGraphRange(backupEntries = null, backupEntriesKey = null) {
```
Comment (Lekuruu) [inline app/static/js/user.js:707]: Consider finding an alternative for `= null`, because uhh.. old browsers :laughing:

**Review by Lekuruu — CHANGES_REQUESTED** [review]

File: app/static/js/user.js (hunk 1 above)
Comment (Lekuruu) [reply app/static/js/user.js:707]: Yep. I am not sure if javascript has the `variable || "default"` thing that python does, but oh well. Works.

**Review by Lekuruu — COMMENTED** [review]
https://github.com/user-attachments/assets/af760963-6f7c-4119-8d6c-45d2e79b1d98

Works fine for most people, but some seem to have these dots on their graph..?

**Comment by Lekuruu** [comment]
Small edge cases. Who knows :laughing: 
I'll test on staging, maybe it won't occur there...

**Comment by Lekuruu** [comment]
You should be able to debug this with this profile: http://osu.staging.titanic.sh/u/183

**Comment by Lekuruu** [comment]
Seems like I am the only one that can get past it, *sigh*...
Maybe you are lucky enough to find an older browser that doesn't have the requirement :laughing:

**Review by Lekuruu — APPROVED** [review]
I'll just merge this to make debugging a little easier for you.

**Comment by Lekuruu** [comment]
Yeah it's working mostly fine. I've noticed that for profiles where the rank has not changed, it will be weirdly.. "dotted"?
